pkg/logql/evaluator.go

```diff
@@ -898,3 +949,31 @@ func literalStepEvaluator(
 		eval.Error,
 	)
 }
+
+// absentLabels guesses labels based on labels matchers of a SampleExpr
+// This is to replace missing timeseries during absent_over_time aggregation.
+func absentLabels(expr SampleExpr) labels.Labels {
+	m := labels.Labels{}
+
+	lm := expr.Selector().Matchers()
+	if len(lm) == 0 {
+		return m
+	}
+
+	empty := []string{}
+	for _, ma := range lm {
+		if ma.Name == labels.MetricName {
+			continue
+		}
+		if ma.Type == labels.MatchEqual && !m.Has(ma.Name) {
+			m = labels.NewBuilder(m).Set(ma.Name, ma.Value).Labels()
+		} else {
```

Comment:
What about `MatchRegexp`? Do you think it makes more sense to leave it as one of the labels to be deleted (which likely won't delete anything)? Otherwise we could treat it as an additive label, but we may see results like `{app="foo.*"}` which isn't great  either as we can't signal `{app=~"foo.*"}` in a returned label set. 

Idk here, it might be best as you have it now.

Reply:
I didn't wanted to do anything different than Prometheus here, I guess they can't predict what didn't match.